climakitaegui/explore/warming.py

```diff
@@ -1088,3 +1088,159 @@ def fit_models_and_plots(
     plt.show()
 
     return new_params, trad_params
+
+
+class IPCCVisualize:
+    """Class for standalone visualization of IPCC warming trajectories
+
+    Attributes
+    ----------
+    none
+
+    """
+
+    def __init__(self):
+        """
+        Read in data for warming trajectory plots
+        """
+        self.ssp119_data = read_csv_file(SSP119_FILE, index_col="Year")
+        self.ssp126_data = read_csv_file(SSP126_FILE, index_col="Year")
+        self.ssp245_data = read_csv_file(SSP245_FILE, index_col="Year")
+        self.ssp370_data = read_csv_file(SSP370_FILE, index_col="Year")
+        self.ssp585_data = read_csv_file(SSP585_FILE, index_col="Year")
+        self.hist_data = read_csv_file(HIST_FILE, index_col="Year")
+
+        # Define SSP mapping
+        self.ssp_mapping = {
+            "SSP 1-1.9": (self.ssp119_data, "#00a9cf", "SSP1-1.9"),
+            "SSP 1-2.6": (self.ssp126_data, "#003466", "SSP1-2.6"),
+            "SSP 2-4.5": (self.ssp245_data, "#f69320", "SSP2-4.5"),
+            "SSP 3-7.0": (self.ssp370_data, "#df0000", "SSP3-7.0"),
+            "SSP 5-8.5": (self.ssp585_data, "#980002", "SSP5-8.5"),
+        }
+
+    def plot_warming_trajectories(
+        self,
+        warming_level: float = 1.5,
+        ssp: str = "All",
+        width: int = 575,
+        height: int = 300,
+    ):
+        """Create visualization of warming trajectories
+
+        Parameters
+        ----------
+        warming_level : float
+            Warming level in degrees Celsius (1.5, 2, 3, or 4)
+        ssp : str
+            Scenario to plot. One of "All", "SSP 1-1.9", "SSP 1-2.6", "SSP 2-4.5",
+            "SSP 3-7.0", "SSP 5-8.5"
+        width : int
+            Plot width in pixels
+        height : int
+            Plot height in pixels
+
+        Returns
+        -------
+        holoviews.core.overlay.Overlay
+            Plot object that can be displayed in notebook
+        """
+        # Plot historical data
+        plot = self.hist_data.hvplot(
+            x="Year",
+            y="Mean",
+            color="k",
+            label="Historical",
+            width=width,
+            height=height,
+        ) * self.hist_data.hvplot.area(
+            x="Year",
+            y="5%",
+            y2="95%",
+            alpha=0.1,
+            color="k",
+            ylabel="°C",
+            xlabel="",
+            ylim=[-1, 5],
+            xlim=[1950, 2100],
+        ).opts(
+            tools=["pan", "wheel_zoom"]
+        )
+
+        # Plot scenarios
+        if ssp == "All":
```

Comment:
The way your statement works here means that if someone passes something that isn't "All" or something in self.ssp_mapping it will just silently skip all this and return the plot anyway. Is that intended behavior? Maybe add an "else" clause that just throws a warning at the user saying their input wasn't meaningful and that they should review?